src/frontend-scripts/components/Gamenotes.jsx

```diff
@@ -82,15 +81,70 @@ class Gamenotes extends React.Component {
 				draggable="true"
 				onDragStart={this.noteDragStart}
 				className="notes-container"
-				style={{ top: `${this.state.top}px`, left: `${this.state.left}px`, height: `${this.state.height}px`, width: `${this.state.width}px` }}
+				style={{
+					top: `${this.state.top}px`,
+					left: `${this.state.left}px`,
+					height: `${this.state.height}px`,
+					width: `${this.state.width}px`
+				}}
 			>
+				{this.state.clearConfirmationShown ? (
+					<div
+						className="notes-container_confirm-container"
+						style={{
+							top: '85px',
+							left: '20px',
+							height: '150px',
+							width: `${this.state.width - 40}px`
+						}}
+					>
+						<div className="notes-container_confirm-header">Confirm Clear?</div>
+						<div className="notes-container_confirm-body">
+							Are you sure you want to clear your notes?
+							<button
+								className="notes-container_confirm-button"
+								style={{ marginRight: '10px' }}
+								onClick={() => {
+									this.clearNotes();
+									this.setState({
+										clearConfirmationShown: false
+									});
+								}}
+							>
+								Yes, Clear
+							</button>
+							<button
+								className="notes-container_confirm-button"
+								onClick={() => {
+									this.setState({
+										clearConfirmationShown: false
+									});
+								}}
+							>
+								No, Cancel
+							</button>
+						</div>
+					</div>
+				) : (
+					<React.Fragment />

```

Comment:
This closing tag appears to be an orphan

Reply:
That's the open and shut tag for a fragment, it's only so that it renders an empty object (i.e. nothin to the screen) if state doesn't call for the confirmation dialog to be shown.

Reply:
oh you don't want a ternary then you just want an `&&`.  Make 91 `{this.state.clearConfirmationShown && (` and delete the negative ternary on 128 and 129